components/application-broker/internal/access/mapping.go

```diff
@@ -30,3 +38,59 @@ func (c *ApplicationMappingService) IsApplicationEnabled(namespace, name string)
 		return false, errors.Wrapf(err, "while getting ApplicationMapping %s/%s", namespace, name)
 	}
 }
+
+// NewServiceChecker creates a ServiceEnabledChecker which is able to check if a service of an application
+// is enabled in the namespace.
+func (c *ApplicationMappingService) NewServiceChecker(namespace, name string) (ServiceEnabledChecker, error) {
+	am, err := c.lister.ApplicationMappings(namespace).Get(name)
+	switch {
+	case err == nil:
+		return newApplicationServiceChecker(am), nil
+	case k8sErrors.IsNotFound(err):
+		return &allServicesDisabled{}, nil
+	default:
+		return nil, errors.Wrapf(err, "while getting ApplicationMapping %s/%s", namespace, name)
+	}
+}
+
+func newApplicationServiceChecker(am *v1alpha1.ApplicationMapping) ServiceEnabledChecker {
+	if am.IsAllApplicationServicesEnabled() {
+		return &allServicesEnabled{}
+	}
+	serviceIDs := make(map[internal.ApplicationServiceID]struct{})
+
+	for _, svc := range am.Spec.Services {
+		serviceIDs[internal.ApplicationServiceID(svc.ID)] = struct{}{}
+	}
+
+	return &ApplicationServiceChecker{
+		serviceIDs: serviceIDs,
+	}
+}
+
+// ApplicationServiceChecker is a default ServiceEnabledChecker implementation
+type ApplicationServiceChecker struct {
+	serviceIDs map[internal.ApplicationServiceID]struct{}
+}
+
+// IsServiceEnabled returns true if the service is enabled
+func (c *ApplicationServiceChecker) IsServiceEnabled(svc internal.Service) bool {
+	_, exists := c.serviceIDs[svc.ID]
+	return exists
+}
+
+type allServicesDisabled struct {
+}
+
+// IsServiceEnabled implements ServiceEnabledChecker, always returns false
+func (c *allServicesDisabled) IsServiceEnabled(svc internal.Service) bool {
+	return false
+}
+
+type allServicesEnabled struct {
+}
+
+// IsServiceEnabled implements ServiceEnabledChecker, always returns true
+func (c *allServicesEnabled) IsServiceEnabled(svc internal.Service) bool {
+	return true
+}
```

Comment:
Is it not easier create one `ApplicationServiceChecker` with extra boolean field instead three different struct? :)

Reply:
I'll do it as private. 3 simple implementation allows to avoid `if` statements inside the IsServiceEnabled implementations